### PR TITLE
Make mpirun a symlink to prun, when available

### PR DESCRIPTION
**Pull request by jjhursey**

### CODE REVIEW

**Comment by jjhursey** [comment]
I'm still looking at a couple more build system configs, then I think this will be ready. So I'll leave it in the draft PR state until my builds finish.

**Comment by jjhursey** [comment]
I tested both:
 * `./autogen.pl` with the submodule in place (and my patch from https://github.com/openpmix/prrte/pull/287)
 * `./autogen.pl --no-prrte`

Both do the right thing. The first will install the symlink, the second doesn't. `make distclean` works in both.

Good to merge whenever you want to. Feel free to squash these down if needed.

File: ompi/tools/mpirun/Makefile.am [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 # Additional copyrights may follow
 #
 # $HEADER$
 #
 
+if OMPI_WANT_PRRTE
+install-exec-hook:
+	(cd $(DESTDIR)$(bindir); rm -f mpirun$(EXEEXT); $(LN_S) prun$(EXEEXT) mpirun$(EXEEXT))
+	(cd $(DESTDIR)$(bindir); rm -f mpiexec$(EXEEXT); $(LN_S) prun$(EXEEXT) mpiexec$(EXEEXT))
+	(cd $(DESTDIR)$(bindir); rm -f oshrun$(EXEEXT); $(LN_S) prun$(EXEEXT) oshrun$(EXEEXT))
```
Comment (rhc54) [inline ompi/tools/mpirun/Makefile.am]: Let's not put oshrun in here - let the oshmem code tree do so if it builds

Comment (rhc54) [reply ompi/tools/mpirun/Makefile.am]: Never mind - I'll take it as-is and then make the changes. Thx!

Comment (jjhursey) [reply ompi/tools/mpirun/Makefile.am]: Sounds good. Thanks

**Pull request merged**